### ISSUE TITLE
[CompareWith] issue

### ISSUE THREAD

**Issue by rkalita** [ng-select/ng-select#582]
### Expected behaviour
Hi guys, I was trying to use **compareWith** parameter in your plugin, but it doesn't work.
Here is one of issue inside my project:
    I want to take parameters from '**ActivatedRoute**' and pass them as default values into **ng-select** controls, but I can't do this, because data for this dropdown is using **numeric** type whereas parameters from queryParams are **string**. Yes, of course I can change type inside control manually but it'll take a lot of time and string of code.
It would be great to have a possibility to configure this feature globally.
Thanks

### Actual behaviour
According to README we have it, but how does it works?
![bs44chd](https://user-images.githubusercontent.com/7390722/40537906-51ea1ef8-603a-11e8-99ea-11334a52b864.png)
I was trying to use it through passing function inside **ng-select** like this
```
<form [formGroup]="form">
          <label>Your first ng-select</label>
          <ng-select formControlName="car_id" [items]="cars"
            bindLabel="name"
            bindValue="id"
            [compareWith]="compareFn"
            placeholder="Select some car">
          </ng-select>
</form>

public compareFn(a, b): boolean {
    return a == b;
}
```
no results
https://stackblitz.com/edit/ng-select-example-rhenac?embed=1&file=app/app.component.ts

### More Info
ng-select version: [2.1.0]

browser: [Google Chrome 66.0.3359.181]

reproducible in demo page: [NO]

**Comment by anjmao** [comment]
Hi @rkalita, I noticed that stackblitz template was not updated and was using old angular version, this is why you get errors in your example, could you try this template https://stackblitz.com/edit/ng-select-example?file=app/app.component.ts ?

**Comment by rkalita** [comment]
Hi @anjmao, thank you for your answer, but I'm afraid that you didn't understand the problem. I've updated your stackblitz where you can check it. In this example we have model(_selectedCity_) with default value that has **string** type whereas array of items(_cities_) contains  { id: **string**, name: **string** }
https://stackblitz.com/edit/ng-select-example-dtfw6a?file=app/app.component.ts

**Comment by anjmao** [comment]
From the code I see that compareWith is used only if bindValue is not specified and you binding to object https://github.com/ng-select/ng-select/blob/master/src/ng-select/items-list.ts#L83

In general it is good practice to have same data types. In your case you can simply convert route query param to number using + sign.

```ts
const id = +params['id'];
```

**Comment by rkalita** [comment]
So as I understand there is no way to use **compareWith**  if you specified _bindValue_, in other words it works only for simple, primitive arrays like **string[]** or **number[]** if I want to get specify value(not an object) in model/control. Am i right?
I have a very big project and in some cases the list of queryParams is big enough and different, and they're not consist just numbers, or strings like ?car=1&country=13, it looks like ?car=1&country=USA. 
Did you imagine how is it painful to change type manually every time? And this is why I'm asking you to make this feature works, because I believe that I'm not alone with this problem.

**Comment by varnastadeus** [comment]
#535 same issue with explanation

**Comment by BioSin** [comment]
@varnastadeus hello! I faced with that problem too, I used all the rest autocomplete component in angular2-6 and I think your implementation is one of the best. 
BUT this behaviour is really weird - because if you will not approve this change, all the developers who faced this problem too will be forced to write 20+ LOC for each widget instance to avoid it.
If you will approve this change it will not brake anything (btw: you don't have any notice that [compareWith] isn't works with [bindValue]), so if somebody needs this function they will just use it, who's not - will not use. 

PS: Thank you a lot for your really good component!
